fillers/eips/eip3860.py

```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
+                "cost to deploy are zero: Then this test case is "
+                "equivalent to that of 'test_exact_intrinsic_gas'."
+            )
+
+        blockchain_test(
             pre=pre,
             post=post,
             blocks=[block],
             genesis_environment=env,
-            tag=f"{initcode.name}_under_execution_gas",
+            tag=f"{initcode.name}_{gas_test_case}",
         )
 
-    """
-    Test case 4: Test with exact execution gas, contract create succeeds.
-    """
-    tx = Transaction(
-        nonce=0,
-        to=None,
-        data=initcode,
-        gas_limit=exact_tx_execution_gas,
-        gas_price=10,
-    )
-    block = Block(txs=[tx])
-    post[created_contract_address] = Account(code=initcode.deploy_code)
-
-    yield BlockchainTest(
-        pre=pre,
-        post=post,
-        blocks=[block],
-        genesis_environment=env,
-        tag=f"{initcode.name}_exact_execution_gas",
-    )
-
-
-@test_from(fork=Shanghai)
-def test_initcode_limit_contract_creating_tx_gas_usage(fork):
-    """
-    Test EIP-3860 Limit Initcode Gas Usage for a contract
-    creating transaction, using different initcode lengths.
-    """
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_MAX_LIMIT,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ONES_MAX_LIMIT,
-        eip_3860_active=True,
-    )
-
-    # Test cases to verify the initcode word cost limits
-
-    yield from generate_gas_cost_test_cases(
-        initcode=EMPTY_INITCODE,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=SINGLE_BYTE_INITCODE,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_32_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_33_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_49120_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_49121_BYTES,
-        eip_3860_active=True,
-    )
-
 
 """
 Test cases using the CREATE opcode
 """
 
 
-def generate_create_opcode_initcode_test_cases(
-    opcode: str,
-    initcode: Initcode,
-    eip_3860_active: bool,
-):
+@pytest.mark.parametrize(
+    "initcode",
+    [
+        INITCODE_ZEROS_MAX_LIMIT,
+        INITCODE_ONES_MAX_LIMIT,
+        INITCODE_ZEROS_OVER_LIMIT,
+        INITCODE_ONES_OVER_LIMIT,
+        EMPTY_INITCODE,
+        SINGLE_BYTE_INITCODE,
+        INITCODE_ZEROS_32_BYTES,
+        INITCODE_ZEROS_33_BYTES,
+        INITCODE_ZEROS_49120_BYTES,
+        INITCODE_ZEROS_49121_BYTES,
+    ],
+    ids=get_initcode_name,
+)
+@pytest.mark.parametrize("opcode", ["create", "create2"], ids=lambda x: x)
```

Comment:
I think we should parametrize here directly using the `Op.CREATE` and `Op.CREATE2`.

Reply:
Definitely more explicit and cleaner.